seqr/views/apis/anvil_workspace_api_tests.py

```diff
@@ -412,40 +452,191 @@ def test_create_project_from_workspace(self, mock_tempfile, mock_mv_file, mock_s
+
+        self._assert_loading_req_exceptions(
+            mock_api_logger, mock_airtable_logger, mock_slack, mock_send_email, project2, "GRCh38",
+            samples=['HG00735', 'NA19675', 'NA19678']
+        )
+
+        # Test logged in locally
+        remove_token(self.manager_user)  # The user will look like having logged in locally after the access token is removed
+        response = self.client.post(url)
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.url, '/login/google-oauth2?next=/api/create_project_from_workspace/my-seqr-billing/anvil-no-project-workspace2/submit')
+
+    @mock.patch('seqr.views.apis.anvil_workspace_api.ANVIL_LOADING_DELAY_EMAIL', None)
+    @responses.activate
+    def test_add_workspace_data(self, mock_send_email, mock_datetime, mock_api_logger, mock_tempfile, mock_mv_file,
+                                mock_slack, mock_load_file, mock_airtable_logger, mock_utils_logger):
+        self._setup_dag_trigger(responses, ADD_DATA_UPDATE_DAG_TASKS_RESP)
+
+        # Test requesting to load data from a workspace without an existing project
+        url = reverse(add_workspace_data, args=['no_PROJECT1_GUID'])
+        self.client.force_login(self.collaborator_user)
+        response = self.client.post(url, content_type='application/json', data=json.dumps({}))
+        self.assertEqual(response.status_code, 404)
+        self.assertEqual(response.json()['error'], 'Project matching query does not exist.')
+
+        # Test insufficient Anvil workspace permission
+        url = reverse(add_workspace_data, args=[PROJECT2_GUID])
+        response = self.client.post(url, content_type='application/json', data=json.dumps({}))
+        self.assertEqual(response.status_code, 403)
+        self.assertEqual(response.json()['error'], 'Permission Denied')
+
+        # Test missing required fields in the request body
+        url = reverse(add_workspace_data, args=[PROJECT1_GUID])
+        self.client.force_login(self.manager_user)
+
+        self._test_missing_required_fields(url, ['uploadedFileId', 'fullDataPath', 'vcfSamples'], TEST_WORKSPACE_NAME)
+
+        self._test_sample_data_error(mock_load_file, url)
+
+        self._test_missing_samples(mock_load_file, url)
+
+        # Test missing loaded samples
+        mock_load_file.return_value = LOAD_SAMPLE_DATA
+        response = self.client.post(url, content_type='application/json', data=json.dumps(REQUEST_BODY))
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(
+            response.json()['error'],
+            'In order to add new data to this project, new samples must be joint called in a single VCF with all previously'
+            ' loaded samples. The following samples were previously loaded in this project but are missing from the VCF:'
+            ' HG00731, HG00732, HG00733, NA19675_1, NA20870, NA20874')
+
+        # Test a valid operation
+        mock_load_file.return_value = LOAD_SAMPLE_DATA
```

Comment:
this is the same as line 526 above, what does assigning it again do?

Reply:
It is redundant. I'll delete it.